### ISSUE TITLE
Allow disabling calls to `console.clear`

### ISSUE THREAD

**Issue by DrewML** [fkling/astexplorer#135]
Frequently when I'm writing a codemod on the site, I have some objects logged out to the console (with some expanded). Every time I forget this and make a change to the codemod, the data I'm looking at gets blown away.

It would be nice if this was an optional that could be disabled in the UI. I'd be happy to do the work if you're open to it.

Having said that, I'd imagine we would need to determine a place in the UI where it was appropriate to show options that aren't specific to the parser. Thoughts?

**Comment by fkling** [comment]
> Having said that, I'd imagine we would need to determine a place in the UI where it was appropriate to show options that aren't specific to the parser. Thoughts?

Yeah, I think that's the biggest issue actually :D Don't really have a good idea yet. Maybe a gear button in the top right of the transformer which opens a settings dialog.


In the meantime, I believe if you select "preserve log" in Chrome, it should not clear the console.

**Comment by DrewML** [comment]
![smh](https://media.giphy.com/media/l2R035GINW2k6ZoxG/giphy.gif)

Can't believe I didn't realize that - I had been making `console.clear` a no-op manually when loading the page. **THANK YOU**

**Comment by DrewML** [comment]
Repurposing the issue to just account for adding a pattern in the UI where non-parser-related settings can be added moving forward.

Examples:

- [Tabs vs Spaces](https://github.com/fkling/astexplorer/issues/116)
- [Disabling clearing console](https://github.com/fkling/astexplorer/issues/135#issue-161798026)

**Comment by hzoo** [comment]
In 6.16 for Babel, https://babeljs.io/blog/2016/09/28/6.16.0#new-feature you can pass down both parser options and generator options. I made the changes and tested them for recast/astexplorer in mind so would be good to have as an option.
